multiqc/modules/sourmash/gather.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse similarity matrix output by sourmash gather """
+
+import csv
+import logging
+from collections import OrderedDict
+
+from multiqc.plots import bargraph
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class GatherMixin:
+    def parse_gather(self):
+        """
+        Modeled after the kraken module and the sourmash compare module.
+        """
+
+        self.top_n = 5
+
+        # find and load gather reports
+        self.gather_raw_data = dict()
+        for f in self.find_log_files("sourmash/gather", filehandles=True):
+            data = []
+            for line in csv.DictReader(f["f"]):
+                data.append(
+                    {
+                        "query_name": line["query_name"],
+                        "match_name": line["name"],
+                        "pct_unique_weighted": float(line["f_unique_weighted"]) * 100,
+                        "pct_match": float(line["f_match"]) * 100,
+                    }
+                )
+            if data:
+                self.gather_raw_data[f["s_name"]] = data
+            self.add_data_source(f, section="gather")
+
+        self.gather_raw_data = self.ignore_samples(self.gather_raw_data)
+
+        if len(self.gather_raw_data) == 0:
+            return 0
+
+        self.write_data_file(self.gather_raw_data, "gather")
```

Comment:
If customising the data filename, should still follow MultiQC conventions.

```suggestion
        self.write_data_file(self.gather_raw_data, "multiqc_sourmash_gather")
```

Reply:
Also, the structure of this data is too complex - if saving TSV, the default (or `-k tsv`), the data can't be excessively nested. Currently, it saves this, which is not terribly helpful:

```tsv
Sample
SRR19888423ass-vs-genbank-2022.03-k31
SRR19888427ass-vs-genbank-2022.03-k31
```

Maybe better to save some of the more processed data further on instead.